napari/layers/_shapes_layer/model.py

```diff
@@ -1251,7 +1348,141 @@ def to_xml_list(self, shape_type=None):
+                if len(slice_key) > 0:
+                    slices = data.to_masks(
+                        mask_shape=mask_shape[-2:], shape_type=shape_type
+                    )
+                    for m in slices:
+                        vol = np.zeros(mask_shape)
+                        vol[slice_key] = m
+                        masks.append(vol)
+        if len(masks) == 0:
+            masks = np.array(masks)
+        else:
+            masks = np.stack(masks, axis=0)
+        return masks
+
+    def to_labels(self, labels_shape=None, shape_type=None):
+        """Returns a integer labels image, where each shape is embedded in an
+        array of shape labels_shape with the value of the index + 1
+        corresponding to it, and 0 for background. Passing a `shape_type`
+        argument leads to only labels from that particular `shape_type` being
+        returned. These labels will be renumbered appropriately. For
+        overlapping shapes z-ordering will be respected.
+
+        Parameters
+        ----------
+        labels_shape : np.ndarray | tuple | None
+            Tuple defining shape of labels image to be generated. If non
+            specified, takes the max of all the vertiecs
+        shape_type : {'line', 'rectangle', 'ellipse', 'path', 'polygon'} |
+                     None, optional
+            String of shape type to be included.
+
+        Returns
+        ----------
+        labels : np.ndarray
+            Integer array where each value is either 0 for background or an
+            integer up to N for points inside the corresponding shape.
+        """
+        if labels_shape is None:
+            labels_shape = self.shape
+
+        if self.broadcast or self.ndim == 2:
+            # For broadcast shapes or 2D shapes convert current view to labels
+            # and broadcast across sliced dimensions
+            labels = self.slice_data.to_labels(
+                labels_shape=labels_shape[-2:], shape_type=shape_type
+            )
+            labels = np.broadcast_to(labels, labels_shape)
+        else:
+            # For nD insert each keyed slice into correct place in volume
+            # and increment integer label of shape
+            labels = np.zeros(labels_shape)
+            nshapes = 0
+            for slice_key, data in self.data.items():
+                if len(slice_key) > 0:
+                    slices = data.to_labels(
+                        labels_shape=labels_shape[-2:], shape_type=shape_type
+                    )
+                    slices[slices > 0] += nshapes
+                    labels[slice_key] = slices
+                    nshapes += len(data.shapes)
+        return labels
+
+    def to_list(self, shape_type=None):
+        """Returns the vertex data assoicated with the shapes as a list
+        where each element of the list corresponds to one shape. Passing a
+        `shape_type` argument leads to only that particular `shape_type`
+        being returned.
```

Comment:
please follow PEP257 format, which is a single line summary (present imperative, "Do this" rather than "Does this"), followed by a blank line, followed by longer summary optionally

Reply:
fixed